tools/walletextension/wallet_extension.go

```diff
@@ -162,8 +160,25 @@ func (we *WalletExtension) handleHTTPEthJSON(resp http.ResponseWriter, req *http
 		return
 	}
 
+	// for obscuro RPC requests it is important we know the sender account for the viewing key encryption/decryption
```

Comment:
feels like all this logic from 163 to 181 belong in a function in the multi_acc_helper